Modules/_multiprocessing/multiprocessing.c

```diff
@@ -188,33 +188,35 @@ multiprocessing_exec(PyObject *module)
 {
 #ifdef HAVE_MP_SEMAPHORE
 
-    /* Add _PyMp_SemLock type to module */
-    if (PyModule_AddType(module, &_PyMp_SemLockType) < 0) {
+    PyTypeObject *semlock_type = (PyTypeObject *)PyType_FromModuleAndSpec(
+                module, &_PyMp_SemLockType_spec, NULL);
+
+    if (semlock_type == NULL) {
         return -1;
     }
-
-    {
-        PyObject *py_sem_value_max;
-        /* Some systems define SEM_VALUE_MAX as an unsigned value that
-         * causes it to be negative when used as an int (NetBSD).
-         *
-         * Issue #28152: Use (0) instead of 0 to fix a warning on dead code
-         * when using clang -Wunreachable-code. */
-        if ((int)(SEM_VALUE_MAX) < (0))
-            py_sem_value_max = PyLong_FromLong(INT_MAX);
-        else
-            py_sem_value_max = PyLong_FromLong(SEM_VALUE_MAX);
-
-        if (py_sem_value_max == NULL) {
-            return -1;
-        }
-        if (PyDict_SetItemString(_PyMp_SemLockType.tp_dict, "SEM_VALUE_MAX",
-                             py_sem_value_max) < 0) {
-            Py_DECREF(py_sem_value_max);
-            return -1;
-        }
+    if (PyModule_AddType(module, semlock_type) < 0) {
+        return -1;
+    }
+    Py_DECREF(semlock_type);
+    PyObject *py_sem_value_max;
+    /* Some systems define SEM_VALUE_MAX as an unsigned value that
+     * causes it to be negative when used as an int (NetBSD).
+     *
+     * Issue #28152: Use (0) instead of 0 to fix a warning on dead code
+     * when using clang -Wunreachable-code. */
+    if ((int)(SEM_VALUE_MAX) < (0))
+        py_sem_value_max = PyLong_FromLong(INT_MAX);
+    else
+        py_sem_value_max = PyLong_FromLong(SEM_VALUE_MAX);
```

Comment:
Since you are touching practically the whole function, please follow PEP-7.

```suggestion
    if ((int)(SEM_VALUE_MAX) < (0)) {
        py_sem_value_max = PyLong_FromLong(INT_MAX);
    }
    else {
        py_sem_value_max = PyLong_FromLong(SEM_VALUE_MAX);
    }
```

Reply:
@erlend-aasland Can we move out repeated `py_sem_value_max = PyLong_FromLong(...)` altogether?

```c
    long sem_value_max = (int)(SEM_VALUE_MAX) < 0 ? INT_MAX : SEM_VALUE_MAX;
    py_sem_value_max = PyLong_FromLong(sem_value_max);
```

Reply:
I'd prefer not to. We tend to not use the implicit `cond ? expr : expr` in the CPython code base; the explicit `if ... else ...` is preferred, mostly because of readability. (That does not mean there is no such condensed oneliners; you'll find them, but they are rare.)